tools/cosmovisor/internal/watchers/http_block_checker.go

```diff
@@ -0,0 +1,91 @@
+package watchers
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"cosmossdk.io/log"
+)
+
+func NewHTTPRPCBLockChecker(baseUrl string, logger log.Logger) HeightChecker {
+	return &httpRPCBlockChecker{
+		baseUrl: baseUrl,
+		logger:  logger,
+	}
+}
+
+type httpRPCBlockChecker struct {
+	baseUrl string
+	subUrl  string
+	logger  log.Logger
+}
+
+func (j *httpRPCBlockChecker) GetLatestBlockHeight() (uint64, error) {
+	if j.subUrl != "" {
+		return j.getLatestBlockHeight(j.subUrl)
+	}
+
+	height, err1 := j.getLatestBlockHeight("/v1/block")
+	if err1 == nil {
+		j.logger.Info("Successfully resolved latest block height from /v1/block", "url", j.baseUrl+"/v1/block")
+		// If we successfully got the height from /v1/block, we can cache the subUrl
+		j.subUrl = "/v1/block"
+		return height, nil
+	}
+
+	height, err2 := j.getLatestBlockHeight("/block")
+	if err2 == nil {
+		j.logger.Info("Successfully resolved latest block height from /block", "url", j.baseUrl+"/block")
+		// If we successfully got the height from /block, we can cache the subUrl
+		j.subUrl = "/block"
+		return height, nil
+	}
```

Comment:
Might be nicer to just specify a `var` which contains all `subUrl`s to check and iterate through them since the logic for each is the same here.